File: gpp-api/src/main/java/com/develop/gpp/domain/entity/PecasEstoqueModel.java
package com.develop.gpp.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "pecas_estoque")
public class PecasEstoqueModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_peca_estoque;

    @Column(nullable = false)
    private Integer filial;

    @Column(nullable = false)
    private Long id_peca;

    @Column(nullable = false)
    private Long id_box;

    @Column(nullable = false)
    private Integer saldo_disponivel;

    @Column(nullable = false)
    private Integer saldo_reservado;

    @Column(nullable = false)
    private Integer quantidade_transferencia;

    @Column(nullable = false)
    private Integer quantidade_minima;

    // @ManyToOne(fetch = FetchType.LAZY)
    // @JoinColumn(name = "id_peca", referencedColumnName = "id_peca", insertable = false, updatable = false)
    // private Pecas peca;

    // @ManyToOne(fetch = FetchType.LAZY)
    // @JoinColumn(name = "id_box", referencedColumnName = "id_box", insertable = false, updatable = false)
    // private BoxEnderecamento box;

    @Column(nullable = false)
    private String fornecedor;

    @Column(nullable = false)
    private String endereco;

    // getters and setters
}

File: angular-gpp/src/app/models/PecasEstoqueModel.ts
export class PecasEstoqueModel {
  id_peca_estoque: number = 0;
  filial: number = 0;
  id_peca: number = 0;
  id_box: number = 0;
  saldo_disponivel: number = 0;
  saldo_reservado: number = 0;
  quantidade_transferencia: number = 0;
  quantidade_minima: number = 0;
  peca: PecasModel = new PecasModel;
  box: BoxEnderecamentoModel = new BoxEnderecamentoModel;
  fornecedor: string = "";
  endereco: string = "";


}
